client/src/components/Workflow/Editor/Forms/FormTool.vue

```diff
@@ -1,32 +1,34 @@
 <template>
     <ToolCard
-        v-if="hasData"
-        :id="configForm.id"
-        :version="configForm.version"
-        :title="configForm.name"
-        :description="configForm.description"
+        :id="configForm?.id || step.content_id"
+        :version="configForm?.version"
+        :title="configForm?.name || step.name"
+        :description="configForm?.description"
```

Comment:
Either you have the config form or you don't. separating this out here is a maintenance problem for anyone working on this in the future, that's why I suggested separating out the tool portion from the step portion.